nvflare/lighter/cc_provision/impl/cc.py

```diff
@@ -0,0 +1,195 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from typing import Any, Dict, Optional, Type
+
+from nvflare.app_opt.confidential_computing.cc_manager import SHUTDOWN_JOB, SHUTDOWN_SYSTEM
+from nvflare.lighter import utils
+from nvflare.lighter.constants import PropKey, TemplateSectionKey
+from nvflare.lighter.ctx import ProvisionContext
+from nvflare.lighter.entity import Participant, Project
+from nvflare.lighter.spec import Builder
+
+from ..cc_constants import CC_AUTHORIZERS_KEY, CCAuthConfig, CCConfigKey, CCConfigValue, CCManagerArgs
+from .mock import MockBuilder
+from .onprem_cvm import OnPremCVMBuilder
+
+JOB_RETURN_CODE_MAPPING = {
+    "stop_system": SHUTDOWN_SYSTEM,
+    "stop_job": SHUTDOWN_JOB,
+}
+
+CC_MGR_PATH = "nvflare.app_opt.confidential_computing.cc_manager.CCManager"
+
+
+# (deploy_env, CPU_CC_MECHANISM, GPU_CC_MECHANISM)
+VALID_COMPUTE_ENVS = [
+    (
+        CCConfigValue.ONPREM_CVM,
+        CCConfigValue.AMD_SEV_SNP,
+        CCConfigValue.NVIDIA_CC,
+    ),
+    (
+        CCConfigValue.MOCK,
+        None,
+        None,
+    ),
+]
+
+
+BUILDER_CLASSES = {
+    CCConfigValue.ONPREM_CVM: OnPremCVMBuilder,
+    CCConfigValue.MOCK: MockBuilder,
+}
+
+
+class CCBuilder(Builder):
+    """Builder that coordinates different CC implementations (AzureCVM, OnPremCVM, etc.).
+
+    Each CC implementation builder handles all participants that use its implementation.
+    This builder also sets up the CCManager component for each participant.
+    """
+
+    def __init__(
+        self,
+        cc_config_key="cc_config",
```

Comment:
This one should be a constant. It would be very confusing if someone changes this value.

Reply:
ok let me change this and enables multiple Authorizers for the OnPremCVMBuilder